dev/release/release_rc.sh

```diff
@@ -65,7 +65,7 @@ fi
 
 rc_hash="$(git rev-list --max-count=1 "${rc_tag}")"
 
-id="apache-iceberg-go-${version}-rc${rc}"
+id="apache-iceberg-go-${version}"
```

Comment:
Why this change? Doesn't the tar file need to have `-rc${rc}` in the name?

Reply:
apache-iceberg-go-${version}-rc${rc}  is the general convention to follow for release candidates. However  [apache-iceberg ](https://dist.apache.org/repos/dist/dev/iceberg/apache-iceberg-1.7.1-rc0/) and [apache-iceberg-rust](https://dist.apache.org/repos/dist/dev/iceberg/apache-iceberg-rust-0.4.0-rc.1/) are doing it this way.

Reply:
I have also checked a few other apache releases and it looks like some follow and some don't. 
I have added a new commit with the tar ball naming as suggested as to keep it consistent with our 0.1.0 release